src/mongoose_async_pools.erl

```diff
@@ -44,6 +46,28 @@ config_spec() ->
 pool_name(HostType, PoolId) ->
     persistent_term:get({?MODULE, HostType, PoolId}).
 
+-spec sync(mongooseim:host_type(), pool_id()) -> term().
+sync(HostType, PoolId) ->
+    Pids = mongoose_async_pools:get_workers(HostType, PoolId),
+    Context = #{what => sync_failed, host_type => HostType, pool_id => PoolId},
+    F = fun(Pid) -> 
+                safely:apply_and_log(gen_server, call, [Pid, sync], Context)
+        end,
+    Results = mongoose_lib:pmap(F, Pids),
+    check_results(Results).
+
+check_results(Results) ->
+    [check_result(Result) || Result <- Results].
+
+check_result({ok, ok}) -> ok;
+check_result({ok, skipped}) -> ok;
+check_result(Other) -> ?LOG_ERROR(#{what => sync_failed, reason => Other}).
+
+-spec get_workers(mongooseim:host_type(), pool_id()) -> [atom()].
+get_workers(HostType, PoolId) ->
+    Pool = pool_name(HostType, PoolId),
+    wpool:get_workers(Pool).
```

Comment:
Ok, I didn't like how the thing before was a hack, but I'm also not a fan of taking the dependency from github instead of hex. Note also one important thing, worker_pool from main contains code that moves plenty of ets tables into persistent_terms, so I wouldn't take that into MIM's master yet, I'd rather leave it in a PR ta be load-tested before merging. I'm sorry, please revert to the previous ugly hack and we'll update this later when the we release the new worker_pool library and load-test it.